sklearn/utils/estimator_checks.py

```diff
@@ -4564,15 +4572,16 @@ def _check_generated_dataframe(name, case, index, outputs_default, outputs_panda
         ) from e
 
 
-def check_set_output_transform_pandas(name, transformer_orig):
+def _check_set_output_transform_dataframe(
+    name,
+    transformer_orig,
+    dataframe_lib,
+    is_supported_dataframe,
+    create_dataframe,
+    assert_frame_equal,
+    context,
```

Comment:
A short docstring could help poor future maintainers, e.g. what is passed as `context`?

Reply:
I added a full docstring in [`538341f` (#27258)](https://github.com/scikit-learn/scikit-learn/pull/27258/commits/538341fae1b2efd362cbc0222798de4802658e0a) and [`ad6d9b6` (#27258)](https://github.com/scikit-learn/scikit-learn/pull/27258/commits/ad6d9b6f79c95cb6994e3e146fdb53683c4dabaa)